### PR TITLE
[docs] Using Firebase: utilize Step component, other small tweaks

### PR DESCRIPTION
**Pull request by Simek**
# Why

Let's use the new Step component in Firebase guide to improve the readability.

# How

Add step component instead named headers, set the header limit for ToC to forth level, apply few working tweaks and break long single lines for editors convenience.

# Test Plan

The changes have been tested by running docs app locally.

# Preview

Screenshot 2022-11-07 174614

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/guides/using-firebase.mdx [hunk 1]
```diff
@@ -15,13 +18,14 @@ There are two different ways you can use Firebase in your projects:
 
 React Native supports both the native SDK and the JS SDK. The following sections will guide you through when to use which SDK and all the configuration steps required to use Firebase in your Expo projects.
 
-## Prerequisites
+### Prerequisites
```
Comment (amandeepmittal) [inline docs/pages/guides/using-firebase.mdx:21]: Not sure I understand the requirement to change the heading level here. I think all sub-sections headers should start in order.

**Review by amandeepmittal — COMMENTED** [review]
Thanks for making these changes 🙏

File: docs/pages/guides/using-firebase.mdx (hunk 1 above)
Comment (Simek) [reply docs/pages/guides/using-firebase.mdx:21]: I just want to make it "a part" of the intro paragraph (which start with page title - `h1`), but have no strong opinion about this change, can revert that.

Comment (Simek) [inline docs/pages/guides/using-firebase.mdx:21]:
```suggestion
## Prerequisites
```

Comment (amandeepmittal) [reply docs/pages/guides/using-firebase.mdx]: In that case, you can remove the heading, "Prerequisites" altogether. I thought it'll be good to mention the obvious but I see your point as well 👍

**Review by amandeepmittal — APPROVED** [review]

**Pull request merged**